composer/utils/dist.py

```diff
@@ -235,23 +236,18 @@ def initialize_dist(backend: str, timeout: datetime.timedelta):
         return
 
     if dist.is_initialized():
-        if not dist.get_backend() == backend.lower():
-            warnings.warn(f"The requested backend ({backend}) differs from the backend "
-                          f"of the current process group ({dist.get_backend()})."
-                          "If you wish to change backends, please restart the python process.")
+        if dist.get_backend() != backend.lower() and get_world_size() > 1:
```

Comment:
This should apply whenever dist is initialized

```suggestion
        if dist.get_backend() != backend.lower():
```